torch/_guards.py

```diff
@@ -165,11 +167,22 @@ def set_export_info(self, guard_type, guarded_class, code_list, obj_weakref):
         self.obj_weakref = obj_weakref
 
 
+class ShapeGuard:
```

Comment:
Make this a frozen dataclass or named tuple. Named tuple might be better as you no longer have to modify the call sites.